pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -354,10 +354,15 @@ func pidNamespaceForPod(pod *v1.Pod) runtimeapi.NamespaceMode {
 
 // namespacesForPod returns the runtimeapi.NamespaceOption for a given pod.
 // An empty or nil pod can be used to get the namespace defaults for v1.Pod.
-func namespacesForPod(pod *v1.Pod) *runtimeapi.NamespaceOption {
-	return &runtimeapi.NamespaceOption{
-		Ipc:     ipcNamespaceForPod(pod),
-		Network: networkNamespaceForPod(pod),
-		Pid:     pidNamespaceForPod(pod),
+func (m *kubeGenericRuntimeManager) namespacesForPod(pod *v1.Pod) (*runtimeapi.NamespaceOption, error) {
+	userNs, err := m.runtimeHelper.GetUserNamespaceMappings(pod)
+	if err != nil {
+		return nil, err
 	}
+	return &runtimeapi.NamespaceOption{
+		Ipc:           ipcNamespaceForPod(pod),
+		Network:       networkNamespaceForPod(pod),
+		Pid:           pidNamespaceForPod(pod),
+		UsernsOptions: userNs,
```

Comment:
differently than the other namespaces, `m.runtimeHelper.GetUserNamespaceMappings(pod)` can fail.

So we must to do it before to handle a possible error.

Reply:
@giuseppe maybe we can call that function `userNamespaceForPod()` instead? Of course, keep the code as it is, as it can return an error.

Reply:
@sftim no, sorry, I strongly oppose to that. We don't need to make a whole bunch of other functions return an error, just because one function that _can_ have a similar name returns an error. That makes the code way more complicated, for no reason (they don't return an error!) and users of those functions need lot of boilerplate code to check for errors that can't ever happen.

Sorry, this is a strong no. I'm open to hear reasons for doing it that way, though. Please share them if you think they are worth the pain of handling errors that can never happen.

Reply:
OK, sure. I'm used to the `?` postfix operator for error checking; Go doesn't have one, and so that extra boilerplate is a good reason not to do it the way I suggested.

Reply:
Oh, good point. Looking at the file locally this name indeed seems to fit better with the current code base.

@sftim I will mark this as resolved, then. Please re-open if you don't agree on keeping it as it is.